configs/templates/hpc-stack-dev/spack.yaml

```diff
@@ -0,0 +1,230 @@
+      version: [1.2.3]
+    gsl-lite:
+      version: [0.37.0]
+    hdf:
+      version: [4.2.15]
+      variants: ~fortran ~netcdf
+    hdf5:
+      version: [1.10.6]
+      variants: +hl +fortran +mpi +threadsafe
+    ip:
+      version: [3.3.3]
+    ip2:
+      version: [1.1.2]
+    jasper:
+      version: [2.0.25]
+    jedi-cmake:
+      version: [1.3.0]
+    jpeg:
+      version: [9.1.0]
+    landsfcutil:
+      version: [2.4.1]
+    libjpeg-turbo:
+      version: [2.1.0]
+    libpng:
+      version: [1.6.37]
+    mapl:
+      version: [2.22.0]
+    met:
+      version: [10.1.1]
+      variants: +python +grib2
+    metplus:
+      version: [4.1.1]
+    nco:
+      version: [5.0.6]
+      variants: ~doc
+    nemsio:
+      version: [2.5.4]
+    nemsiogfs:
+      version: [2.5.3]
+    nccmp:
+      version: [1.9.0.1]
+    ncio:
+      version: [1.1.2]
+    netcdf-c:
+      version: [4.7.4]
+      variants: +dap +mpi +parallel-netcdf
+    netcdf-cxx4:
+      version: [4.3.1]
+    netcdf-fortran:
+      version: [4.5.3]
+    nlohmann-json:
+      version: [3.10.5]
+    nlohmann-json-schema-validator:
+      version: [2.1.0]
+    openblas:
+      version: [0.3.19]
+      variants: +noavx512
+    parallelio:
+      version: [2.5.7]
+      variants: +pnetcdf
+    parallel-netcdf:
+      version: [1.12.2]
+    pkgconf:
+      buildable: False
+    prod-util:
+      version: [1.2.2]
+    proj:
+      version: [8.1.0]
+      variants: ~tiff
+    python:
+      version: [3.9.12]
+    py-cartopy:
+      variants: +plotting
+    py-click:
+      version: [8.0.3]
+    py-eccodes:
+      version: [1.3.2]
+    py-f90nml:
+      version: [1.4.2]
+    py-h5py:
+      version: [3.6.0]
+    py-netcdf4:
+      version: [1.5.3]
+      variants: +mpi
+    py-numpy:
+      version: [1.22.3]
+      variants: +blas +lapack
+    py-openpyxl:
+      version: [3.0.3]
+    py-pandas:
+      version: [1.4.0]
+    py-pybind11:
+      version: [2.8.1]
+    py-pycodestyle:
+      version: [2.8.0]
+    py-pygithub:
+      version: [1.55]
+    py-pygrib:
+      version: [2.1.4]
+    py-pyhdf:
+      version: [0.10.4]
+    py-pyproj:
+      version: [3.1.0]
+    py-python-dateutil:
+      version: [2.8.2]
+    py-pythran:
+      version: [0.11.0]
+    py-pyyaml:
+      version: [6.0]
+    py-scipy:
+      version: [1.8.0]
+    py-shapely:
+      version: [1.8.0]
+    sfcio:
+      version: [1.4.1]
+    shumlib:
+      version: [macos_clang_linux_intel_port]
+    sigio:
+      version: [2.3.2]
+    sp:
+      version: [2.3.3]
+    udunits:
+      version: [2.2.28]
+    upp:
+      version: [10.0.10]
+    w3emc:
+      version: [2.9.2]
+    w3nco:
+      version: [2.4.1]
+    wget:
+      version: [1.21.2]
+    wgrib2:
+      version: [3.1.1]
+    wrf-io:
+      version: [1.2.0]
+    yafyaml:
+      version: [0.5.1]
+    zlib:
+      version: [1.2.12]
+
+  definitions:
+  # Set compilers here
+  - compilers: ['%intel']
+  - packages:
+      - global-workflow-env@hpc-stack-dev
```

Comment:
I didn't know that this syntax works, interesting.

Reply:
The matrix or the version? It's useful to declare compilers as a list. If you want to build with multiple compilers you can just add to the list `spack add -l compilers "%gcc"`.

Reply:
I was referring to the `@hpc-stack-dev` syntax, haven't seen that before.

Reply:
The bundles have no source associated with their version so you can give whatever version you want. I think it's good to differentiate bundle versions per environment.

Reply:
Yes, makes total sense.